src/loaders/jobs/taskfarmer/taskfarmer_task_mgr.py

```diff
@@ -0,0 +1,316 @@
+
+        tasks_df = df[(df["kbase_collection"] == self.kbase_collection) &
+                      (df["load_ver"] == self.load_ver) &
+                      (df["tool"] == self.tool)]
+
+        tasks_df.sort_values(by="job_submit_time", ascending=False, inplace=True)
+
+        return tasks_df
+
+    def _create_job_dir(self, destroy_old_job_dir=False):
+        """
+        Create the job directory. If destroy_old_job_dir is True, recreate the job directory.
+        """
+
+        if os.path.exists(self.job_dir) and destroy_old_job_dir:
+            print(f'removing job dir {self.job_dir}')
+            shutil.rmtree(self.job_dir, ignore_errors=True)
+
+        os.makedirs(self.job_dir, exist_ok=True)
+
+        return self.job_dir
+
+    def _get_job_status_from_nersc(self, job_id):
+        """
+        Get the job status from NERSC using squeue command.
+
+        A 'done.tasks.txt.tfin' file is created in the job directory when the job is finished successfully.
+
+        When job is queued, squeue command will return code 0 and out put will be like:
+        JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+        6049897 regular_m submit_t      tgu PD       0:00      2 (Priority)
+
+        When job is running, squeue command will return code 0 and out put will be like:
+        JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+        5987779 regular_m submit_t      tgu  R       1:20      2 nid[005689-005690]
+
+        When job is finished/failed:
+            * job is cached for a period of time (about 10 mins). squeue command will return code 0 and
+            out put will be like below (with no job info just the header):
+            JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+
+            * after the period of time, squeue command will return error code 1 and error message:
+            slurm_load_jobs error: Invalid job id specified
+
+        When job is cancelled:
+            * during cancellation, squeue command will return code 0 and out put will be like:
+            JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+            6050963 regular_m submit_t      tgu CG       1:17      2 nid[004477-004478]
+            * after cancellation, job is cached for a period of time (about 10 mins).
+            squeue command will return code 0 and out put will be like below (with no job info just the header):
+            JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+            * after the period of time, squeue command will return error code -1 and error message:
+            slurm_load_jobs error: Invalid job id specified
+
+        Matrix of job status:
+        Done file  Exit Code   Job listed    Job Status Code  Returned Status
+        Y          Any         N/A           N/A              COMPLETED
+        N          0           N             PD               PENDING
```

Comment:
```suggestion
        N          0           Y             PD               PENDING
```
Pretty sure that's right, let me know if I'm off base here

Reply:
👍

Reply:
:+1: